src/physical/columnar/operations/columnscan_prune.rs

```diff
@@ -0,0 +1,87 @@
+use crate::physical::columnar::traits::columnscan::ColumnScanCell;
+use crate::physical::datatypes::ColumnDataType;
+use crate::physical::tabular::operations::triescan_prune::SharedTrieScanPruneState;
+use std::fmt::Debug;
+use std::ops::Range;
+
+use super::super::traits::columnscan::ColumnScan;
+
+/// Forwards calls to this column though it's [`SharedTrieScanPruneState`] to the input trie.
+/// See `TrieScanPrune` for more information.
+#[derive(Debug)]
+pub struct ColumnScanPrune<'a, T>
+where
+    T: 'a + ColumnDataType,
+{
+    state: SharedTrieScanPruneState<'a>,
+    /// Index this column scan corresponds to in the input trie
+    column_scan_index: usize,
+    reference_scan: &'a ColumnScanCell<'a, T>,
+}
+
+impl<'a, T> ColumnScanPrune<'a, T>
+where
+    T: 'a + ColumnDataType,
+{
+    /// Constructs a new column scan. This should not be called directory, but only by `TrieScanPrune`
+    pub fn new(
+        state: SharedTrieScanPruneState<'a>,
+        column_scan_index: usize,
+        reference_scan: &'a ColumnScanCell<'a, T>,
+    ) -> Self {
+        Self {
+            state,
+            column_scan_index,
+            reference_scan,
+        }
+    }
+}
+
+impl<'a, T> Iterator for ColumnScanPrune<'a, T>
+where
+    T: 'a + ColumnDataType,
+{
+    type Item = T;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        unsafe { (*self.state.get()).get_next_value(self.column_scan_index) };
```

Comment:
Since you have `&mut self` couldn't you do `self.state.get_mut().get_next_value(...)` without the `unsafe` block? Maybe I'm missing something.